pkg/controller/oam/applicationconfiguration/apply.go

```diff
@@ -66,8 +66,15 @@ func (a *workloads) Apply(ctx context.Context, w []Workload, ao ...resource.Appl
 				Name:       wl.Workload.GetName(),
 			}
 			// TODO(rz): Need to find a way to make sure that this is compatible with the trait structure.
-			if err := fieldpath.Pave(t.UnstructuredContent()).SetValue("spec.workloadRef", ref); err != nil {
-				return errors.Wrapf(err, errFmtSetWorkloadRef, t.GetName(), wl.Workload.GetName())
+			spec, ok := t.UnstructuredContent()["spec"].(map[string]interface{})
```

Comment:
we need to examine the CRD not the CR of the trait

Reply:
Got it.